theme/templates/email/signup_verify.txt

```diff
@@ -0,0 +1,8 @@
+{% load i18n %}
+{% block main %}
+{% trans "Welcome to HydroShare. This email address was used to request an account on HydroShare." %}
+{% trans "If you originated the request, please use the link below to verify your email address and activate your account." %}
+http://{{ request.get_host }}{{ verify_url }}
```

Comment:
Do you have the `request` object available in this template? if you do, or if you have `settings` available, you can use HTTP or HTTPS based on the setting or based on the incoming request.